### PR TITLE
feat(cdk): support `SafeHtml` input value in `tui-svg`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/core/components/svg/svg.component.ts [hunk 1]
```diff
@@ -62,39 +62,53 @@ export class TuiSvgComponent {
...
+                ? this.icon
+                : this.resolveName(this.icon, this.iconsPath);
+        }
+
+        return ``;
     }
 
     get isInnerHTML(): boolean {
-        return this.isSrc || this.isExternal || (this.isName && this.isShadowDOM);
+        return typeof this.icon === `string`
```
Comment (waterplea) [inline projects/core/components/svg/svg.component.ts:109]: `return !(typeof this.icon === `string`) || this.isSrc || this.isExternal || (this.isName && this.isShadowDOM)`

File: projects/core/components/svg/svg.component.ts [hunk 2]
```diff
@@ -112,11 +126,11 @@ export class TuiSvgComponent {
     }
 
     private get isUrl(): boolean {
-        return this.icon.endsWith(`.svg`);
+        return typeof this.icon === `string` ? this.icon.endsWith(`.svg`) : false;
```
Comment (waterplea) [inline projects/core/components/svg/svg.component.ts:129]: `return typeof this.icon === 'string' && this.icon.endsWith('.svg')`

File: projects/core/components/svg/svg.component.ts [hunk 3]
```diff
@@ -112,11 +126,11 @@ export class TuiSvgComponent {
...
     }
 
     private get isSrc(): boolean {
-        return tuiIsPresumedHTMLString(this.icon);
+        return typeof this.icon === `string` ? tuiIsPresumedHTMLString(this.icon) : false;
```
Comment (waterplea) [inline projects/core/components/svg/svg.component.ts:133]: `return typeof this.icon === 'string' && tuiIsPresumedHTMLString(this.icon)`

**Review by waterplea — CHANGES_REQUESTED** [review]
Also please use utility function `tuiIsString`.

File: projects/core/components/svg/svg.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/svg/svg.component.ts]: fixed

File: projects/core/components/svg/svg.component.ts (hunk 3 above)
Comment (splincode) [reply projects/core/components/svg/svg.component.ts]: fixed

File: projects/core/components/svg/svg.component.ts (hunk 2 above)
Comment (splincode) [reply projects/core/components/svg/svg.component.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**